### PR TITLE
S7_sorting

### PR DESCRIPTION
**Pull request by valntyn**
[DEMO](https://valntyn.github.io/products-e-commerce/#/products)

### CODE REVIEW

File: src/components/Sort/SortLink/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+type PropTypes = {
+  sort: string;
+  title: string;
+};
+
+export const SortLink: React.FC<PropTypes> = ({ title, sort }) => {
+  const [searchParams] = useSearchParams();
+  const isCurrentSort = searchParams.get('sort') === sort;
+  const isReversed = searchParams.get('order') === 'desc';
```
Comment (Yuliia143) [inline src/components/Sort/SortLink/index.tsx]: Enum ?

File: src/components/Sort/SortLink/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  const dispatch = useAppDispatch();
```
Comment (Yuliia143) [inline src/components/Sort/SortLink/index.tsx:22]: It's better to declare useAppDispatch before search params

File: src/components/Sort/SortLink/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    if (!isCurrentSort) {
+      return getSearchWith(searchParams, {
+        sort: sortBy,
+        order: null,
+      });
+    }
+
+    if (!isReversed) {
```
Comment (Yuliia143) [inline src/components/Sort/SortLink/index.tsx]: Why first 'if' use just condition , but second/third - variables?

File: src/components/Sort/SortLink/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      return getSearchWith(searchParams, { order: 'desc' });
+    }
+
+    return undefined;
```
Comment (Yuliia143) [inline src/components/Sort/SortLink/index.tsx]: ?

File: src/components/Sort/index.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    document.addEventListener('click', handleDocumentClick);
+
+    return () => {
+      document.removeEventListener('click', handleDocumentClick);
+    };
+  }, []);
+
+  useEffect(() => {
+    const sortInParams = searchParams.get('sort') || null;
```
Comment (Yuliia143) [inline src/components/Sort/index.tsx]: Why || null ? Won't undefined be false for the condition below ?

File: src/store/selectors/selectFilteredProducts.ts [hunk 6]
```diff
@@ -34,15 +37,15 @@ export const selectFilteredProducts = createSelector(
...
+          && !selectedRating.includes(product.rating.toString())
         ) {
           return false;
         }
 
         if (
-          selectedPrice[0] !== 0 &&
-          (fixedPrice < selectedPrice[0] || fixedPrice > selectedPrice[1])
+          selectedPrice[0] !== 0
+          && (fixedPrice < selectedPrice[0] || fixedPrice > selectedPrice[1])
```
Comment (Yuliia143) [inline src/store/selectors/selectFilteredProducts.ts]: It's better to create variables for conditions to have better understanding what they mean

**Review by Yuliia143 — CHANGES_REQUESTED** [review]

File: src/components/Sort/SortLink/index.tsx (hunk 2 above)
Comment (valntyn) [reply src/components/Sort/SortLink/index.tsx:22]: I'll remember, fixed it.

File: src/components/Sort/SortLink/index.tsx (hunk 3 above)
Comment (valntyn) [reply src/components/Sort/SortLink/index.tsx]: I changed the structure of the function, now it looks better. I'll remember to take out logic in variables.

File: src/components/Sort/index.tsx (hunk 5 above)
Comment (valntyn) [reply src/components/Sort/index.tsx]: fixed it

File: src/store/selectors/selectFilteredProducts.ts (hunk 6 above)
Comment (valntyn) [reply src/store/selectors/selectFilteredProducts.ts]: I'll remember, made it better.

File: src/components/Sort/SortLink/index.tsx (hunk 1 above)
Comment (valntyn) [reply src/components/Sort/SortLink/index.tsx]: changed type for sort in props to enum

File: src/components/Sort/SortLink/index.tsx (hunk 4 above)
Comment (valntyn) [reply src/components/Sort/SortLink/index.tsx]: I have my linter configured in a way that doesn't allow me to create a function that doesn't return anything. I improved this function and will remember.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**